packages/at_secondary_server/lib/src/verb/handler/lookup_verb_handler.dart

```diff
@@ -55,106 +55,228 @@ class LookupVerbHandler extends AbstractVerbHandler {
-          } on DataStoreException catch (e) {
-            logger.severe('Hive error adding to access log:${e.toString()}');
-          }
-        }
-        return;
-      } else {
-        response.data = null;
+    }
+  }
+
+  /// Retrieve data owned by a different 'atSign':
+  ///
+  ///  - If a cached key is available, fetch data from the cache. However,
+  ///    if [byPassCacheStr] is set to true, the value from cached key is ignored.
+  ///    Initiates a network call to the "atSign" who owns the data and returns the value.
+  ///  - If cached key does not exist, fetches data from the "atSign" that owns the data.
+  Future<void> _fetchDataOwnedByOtherAtSign(
+      String thisServersAtSign,
+      String keyAtAtSign,
+      Response response,
+      String? operation,
+      String? byPassCacheStr) async {
+    String cachedKeyName = '$CACHED:$thisServersAtSign:$keyAtAtSign';
+    //Get cached value.
+    AtData? cachedValue =
+        await cacheManager.get(cachedKeyName, applyMetadataRules: true);
+    response.data = SecondaryUtil.prepareResponseData(operation, cachedValue);
+
+    //If cached value is null or byPassCache is true, do a remote lookUp
+    if (response.data == null ||
+        response.data == '' ||
+        byPassCacheStr == 'true') {
+      AtData? atData =
+          await cacheManager.remoteLookUp(cachedKeyName, maintainCache: true);
+      if (atData != null) {
+        response.data = SecondaryUtil.prepareResponseData(operation, atData,
+            key: '$thisServersAtSign:$keyAtAtSign');
       }
     }
   }
 
+  /// Retrieve data owned by [thisServersAtSign].
+  Future<void> _fetchDataOwnedByThisAtSign(
+      String thisServersAtSign,
+      String keyAtAtSign,
+      InboundConnection atConnection,
+      Response response,
+      String? operation,
+      AtAccessLog? atAccessLog,
+      String keyOwnersAtSign) async {
+    var lookupKey = '$thisServersAtSign:$keyAtAtSign';
+    var lookupValue = await keyStore.get(lookupKey);
+    response.data = SecondaryUtil.prepareResponseData(operation, lookupValue);
+    //Resolving value references to correct value
+    if (response.data != null && response.data!.contains(AT_VALUE_REFERENCE)) {
+      response.data = await resolveValueReference(
+          response.data.toString(), thisServersAtSign);
+    }
+    try {
+      await atAccessLog!
+          .insert(keyOwnersAtSign, lookup.name(), lookupKey: keyAtAtSign);
+    } on DataStoreException catch (e) {
+      logger.severe('Hive error adding to access log:${e.toString()}');
+    }
+  }
+
+  /// When an 'atSign' is authenticated via APKAM, retrieves enrollment data based
+  /// on the enrollmentId.
+  ///
+  /// If the user possesses authorization for the [lookupKey] namespace, returns true;
+  /// otherwise, returns false.
+  Future<bool> _isAuthorizedToViewData(
+      InboundConnection atConnection, String lookupKey) async {
+    // If a key does not have namespace then the client is authorized
```

Comment:
Please add some further explanation of why this is the case.

Reply:
Updated the code comments. Please review.